src/main/kotlin/com/builzer/backend/global/config/feign/GithubClientConfig.kt

```diff
@@ -0,0 +1,29 @@
+package com.builzer.backend.global.config.feign
+
+import com.builzer.backend.global.config.security.MemberPayload
+import feign.RequestInterceptor
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+import org.springframework.http.HttpHeaders
+import org.springframework.security.core.context.SecurityContextHolder
+
+@Profile("!test")
+@Configuration
+class GithubClientConfig {
+    @Bean
+    fun headerRequestInterceptor(): RequestInterceptor {
+        return RequestInterceptor { request ->
+            request.header("Accept", "application/vnd.github.v3+json")
+            request.header("X-GitHub-Api-Version", "2022-11-28")
+
+            val authentication = SecurityContextHolder.getContext().authentication
+
+            if (authentication != null) {
+                val memberPayload: MemberPayload = authentication.principal as MemberPayload
```

Comment:
(may) 강제치환보다는 if (principal is MemberPayload) 처럼 타입 체크 후 꺼내는게 좋아보입니다. 코틀린은 if로 타입체크를 하면 if문안에서 저절로 타입이 캐스팅 됩니다.

Reply:
principal 자체는 Object 타입이라, 강제 캐스팅해야 합니다.
(Smart cast to 'MemberPayload' is impossible, because 'authentication.principal' is a property that has open or custom getter)